app/controllers/concerns/api/compatibility_checker.rb

```diff
@@ -9,5 +9,11 @@ def check_create_host_nested
         Foreman::Deprecation.api_deprecation_warning("Field host_parameters_attributes.nested ignored") unless attribute.delete(:nested).nil?
       end
     end
+
+    def prevent_attributes
```

Comment:
@dLobatog I have made this method generic, so that in future, if we have more attributes like 'capabilities' that needs to be taken care. We can add that attribute as a part of the arrary. Hence named it as prevent_attributes.

Any thoughts?